### ISSUE TITLE
tools: semi-automatic removal of already released changelog entries

### ISSUE THREAD

**Issue by Totktonada** [tarantool/tarantool#8256]
Each user visible change adds a [changelog entry](https://github.com/tarantool/tarantool/blob/master/doc/changelogs.md) into the repository. When a release or a pre-release is created, those entries are merged into a release notes document. The old entries are removed at the release point.

Tarantool development goes in parallel long-lived branches. For example, we have `1.10`, `2.10` and `master` (future `2.11`) at the moment of writing the issue.

The new changes are usually developed on top of `master` and spread down if applicable (usually when it is a fix of a bug that affects lower versions or when it is a general infrastructure change). So the same changelog entries are present in the parallel branches.

Let's assume that we're releasing 2.10.5 and then 2.11.0-rc1. Intuitively a reader expects that the 2.11.0-rc1 release notes lists changes comparing to 2.10.5.

It means that should remove changelog entries for changes that were released in a lower version.

It is not a task that may be done entirely automatic, but we can automatically detect 'cherry picked from commit' marks, same commit subjects, same changelog entry filenames and report all the information to a human to review and decide about an entry removal.

**Comment by Totktonada** [comment]
Duplicate of #9135 and it is now considered solved by PR #9154.

There are known areas for improvements, but since the changelogs cleanup involves much less manual work now, I'm going to close this issue.
